app/src/main/java/space/taran/arknavigator/mvp/presenter/GalleryPresenter.kt

```diff
@@ -40,7 +40,7 @@ class GalleryPresenter(
         viewState.init(previews)
     }
 
-    fun deleteResource(resource: ResourceId) {
+    fun deleteResource(resource: ResourceId) = presenterScope.launch {
```

Comment:
Am I right that coroutines launched in `presenterScope` might be cancelled?
For instance, when the presenter is destroyed.

Would this lead to a half-deleted resource in this method?

Reply:
You are right, coroutines launched in `presenterScope` will be canceled if the presenter is destroyed.
I might be wrong, code execution can be stopped when suspend functions are called, because they can throw a `CancellationException` after execution. 

In this case, `storage.remove(resource)` will be fully executed, but this code will be canceled:
```
val path = index.remove(resource)
Log.d(GALLERY_SCREEN, "path $ path removed from index")
Files.delete(path)
```

There are 2 possible solutions:
- Use a different scope for important work. For example `appScope`, which will be tied to the app lifecycle. 
- Mark important work as `presenterScope.launch(NonCancellable)`

Reply:
`NonCancellable` sounds as easier way, what are pros/cons of these 2 options?

Why `storage.remove(resource)` will be fully executed? Is it because it's coroutine is in another scope?

Reply:
I have too little experience with coroutines, I can't say the pros and cons. Google recommends using a different scope.
https://developer.android.com/kotlin/coroutines/coroutines-best-practices#create-coroutines-data-layer